src/test/java/org/rulez/demokracia/pdengine/VoteCastTest.java

```diff
@@ -0,0 +1,171 @@
+package org.rulez.demokracia.pdengine;
+
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.rulez.demokracia.pdengine.annotations.tested_behaviour;
+import org.rulez.demokracia.pdengine.annotations.tested_feature;
+import org.rulez.demokracia.pdengine.annotations.tested_operation;
+import org.rulez.demokracia.pdengine.dataobjects.CastVote;
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+import org.rulez.demokracia.pdengine.testhelpers.CreatedDefaultChoice;
+
+public class VoteCastTest extends CreatedDefaultChoice {
+
+	@Before
+	public void setUp() throws ReportedException {
+		super.setUp();
+	}
+
+	@tested_feature("Vote")
+	@tested_operation("Cast vote")
+	@tested_behaviour("records cast vote with the vote and user's proxy id")
+	@Test
+	public void cast_vote_records_the_proxy_id() {
+		String ballot = voteManager.obtainBallot(adminInfo.voteId, adminInfo.adminKey);
```

Comment:
there are duplicates of the following couple of lines all around. shift-alt-m is your friend